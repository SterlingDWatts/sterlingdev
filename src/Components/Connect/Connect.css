.Connect {
  height: calc(100vh - 56px);
  min-height: 526px;
  background-color: #00685b;
}

.Connect.Section {
  padding: 0;
}

#connect {
  position: relative;
  top: -56px;
  visibility: hidden;
}

.Connect h2,
.Connect a {
  color: #ffff6b;
}

.Connect--picture-container {
  height: 60%;
}

.Connect--picture {
  height: 100%;
  width: 100%;
  background-image: url(https://sterlingdwatts.github.io/portfolio/images/sterling-colorado.jpg);
  background-size: cover;
  background-position: center;
}

.Connect--info {
  height: 40%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.Connect--info-grid {
  padding: 0 16px;
  display: grid;
  grid-template-columns: 67px 200px;
  column-gap: 16px;
  row-gap: 16px;
  text-align: left;
  line-height: 1;
  color: white;
}

@media screen and (min-width: 500px) {
  .Connect--picture {
    width: 80%;
    margin: 0 auto;
    background-position: 0 20%;
    box-shadow: 0 1px 10px 3px rgba(0, 0, 0, 0.15),
      0 1px 4px 1px rgba(0, 0, 0, 0.15);
  }
}

@media screen and (min-width: 900px) {
  .Connect {
    display: flex;
    flex-direction: row;
  }

  .Connect--info,
  .Connect--picture-container {
    width: 50%;
    height: 100%;
  }

  .Connect--picture {
    width: 100%;
    box-shadow: none;
    background-image: url(https://sterlingdwatts.github.io/portfolio/images/sterling-colorado-big.jpg);
    background-position: top;
  }
}
