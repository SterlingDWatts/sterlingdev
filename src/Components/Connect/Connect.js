import React from "react";
import Section from "../Section/Section";
import "./Connect.css";

function Connect() {
  return (
    <Section className="Connect">
      <div className="Connect--picture-container">
        <div className="Connect--picture"></div>
      </div>
      <div className="Connect--info">
        <div id="connect"> </div>
        <header>
          <h2>Connect</h2>
        </header>
        <div className="Connect--info-grid">
          <span>Cell</span>
          <span>858.609.9463</span>
          <span>Email</span>
          <span>
            <a
              href="mailto:SterlingDWatts@gmail.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              SterlingDWatts@gmail.com
            </a>
          </span>
          <span>Github</span>
          <span>
            <a
              href="https://github.com/SterlingDWatts"
              rel="noopener noreferrer"
              target="_blank"
            >
              /SterlingDWatts
            </a>
          </span>
          <span>Linkedin</span>
          <span>
            <a
              href="https://www.linkedin.com/in/sterlingdwatts/"
              rel="noopener noreferrer"
              target="_blank"
            >
              /sterlingdwatts
            </a>
          </span>
        </div>
      </div>
    </Section>
  );
}

export default Connect;
