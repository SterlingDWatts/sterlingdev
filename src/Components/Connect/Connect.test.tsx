import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Connect from "./Connect";

describe("Connect Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Connect />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Connect />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
