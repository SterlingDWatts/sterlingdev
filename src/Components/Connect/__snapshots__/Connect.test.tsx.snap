// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect Component renders the UI as expected 1`] = `
<section
  className="Connect Section"
>
  <div
    className="Connect--picture-container"
  >
    <div
      className="Connect--picture"
    />
  </div>
  <div
    className="Connect--info"
  >
    <div
      id="connect"
    >
       
    </div>
    <header>
      <h2>
        Connect
      </h2>
    </header>
    <div
      className="Connect--info-grid"
    >
      <span>
        Cell
      </span>
      <span>
        858.609.9463
      </span>
      <span>
        Email
      </span>
      <span>
        <a
          href="mailto:SterlingDWatts@gmail.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          SterlingDWatts@gmail.com
        </a>
      </span>
      <span>
        Github
      </span>
      <span>
        <a
          href="https://github.com/SterlingDWatts"
          rel="noopener noreferrer"
          target="_blank"
        >
          /SterlingDWatts
        </a>
      </span>
      <span>
        Linkedin
      </span>
      <span>
        <a
          href="https://www.linkedin.com/in/sterlingdwatts/"
          rel="noopener noreferrer"
          target="_blank"
        >
          /sterlingdwatts
        </a>
      </span>
    </div>
  </div>
</section>
`;
